src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,66 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 import model.Candidate;
+import model.Cat;
 import model.People;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(n -> Arrays.stream(n.split(","))
+                        .map(Integer::parseInt)
+                        .filter(number -> number % 2 == 0)
+                        .min(Integer::compare))
+                .flatMap(Optional::stream)
+                .min(Integer::compare).orElseThrow(()
+                        -> new RuntimeException("Can't get min value from list: " + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        double average = numbers.stream()
+                .mapToDouble(n -> numbers.indexOf(n) % 2 != 0 ? n - 1 : n)
+                .filter(n -> n % 2 != 0)
+                .average()
+                .getAsDouble();
+        return (double) Math.round(average);
     }
 
-    /**
-     * Given a List of `People` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new People(«Victor», 16, Sex.MAN),
-     * new People(«Helen», 42, Sex.WOMEN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<People> selectMenByAge(List<People> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge && p.getAge() <= toAge)
+                .filter(p -> p.getSex().equals(People.Sex.MAN))
```

Comment:
це все можна в один фільтр вмістити
```suggestion
                .filter(p -> p.getAge() >= fromAge
                        && p.getAge() <= toAge
                        && p.getSex() == People.Sex.MAN)
```